common/element_handle.go

```diff
@@ -296,85 +300,88 @@ func (h *ElementHandle) click(p *Position, opts *MouseClickOptions) error {
 }
 
 func (h *ElementHandle) clickablePoint() (*Position, error) {
-	var quads []dom.Quad
-	var layoutViewport *cdppage.LayoutViewport
-	var err error
-
-	action := dom.GetContentQuads().
-		WithObjectID(h.remoteObject.ObjectID)
-	if quads, err = action.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
-		return nil, fmt.Errorf("cannot request node content quads %T: %w", action, err)
+	var (
+		quads []dom.Quad
+		err   error
+	)
+	getContentQuads := dom.GetContentQuads().WithObjectID(h.remoteObject.ObjectID)
+	if quads, err = getContentQuads.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
+		return nil, fmt.Errorf("cannot request node content quads %T: %w", getContentQuads, err)
 	}
 	if len(quads) == 0 {
 		return nil, fmt.Errorf("node is either not visible or not an HTMLElement: %w", err)
 	}
 
-	action2 := cdppage.GetLayoutMetrics()
-	if layoutViewport, _, _, _, _, _, err = action2.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
-		return nil, fmt.Errorf("cannot get page layout metrics %T: %w", action, err)
+	// Filter out quads that have too small area to click into.
+	var layoutViewport *cdppage.LayoutViewport
+	getLayoutMetrics := cdppage.GetLayoutMetrics()
+	if _, _, _, layoutViewport, _, _, err = getLayoutMetrics.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
+		return nil, fmt.Errorf("cannot get page layout metrics %T: %w", getLayoutMetrics, err)
 	}
 
-	// Filter out quads that have too small area to click into.
-	clientWidth := layoutViewport.ClientWidth
-	clientHeight := layoutViewport.ClientHeight
+	return filterQuads(layoutViewport.ClientWidth, layoutViewport.ClientHeight, quads)
+}
+
+func filterQuads(viewportWidth, viewportHeight int64, quads []dom.Quad) (*Position, error) {
 	var filteredQuads []dom.Quad
 	for _, q := range quads {
 		nq := q
 		for i := 0; i < len(q); i += 2 {
-			nq[i] = math.Min(math.Max(q[i], 0), float64(clientWidth))
-			nq[i+1] = math.Min(math.Max(q[i+1], 0), float64(clientHeight))
+			nq[i] = math.Min(math.Max(q[i], 0), float64(viewportWidth))
+			nq[i+1] = math.Min(math.Max(q[i+1], 0), float64(viewportHeight))
 		}
-
 		// Compute sum of all directed areas of adjacent triangles
 		// https://en.wikipedia.org/wiki/Polygon#Area
-		var area float64 = 0
+		var area float64
 		for i := 0; i < len(q); i += 2 {
 			p2 := (i + 2) % (len(q) / 2)
 			area += (q[i]*q[p2+1] - q[p2]*q[i+1]) / 2
 		}
-
-		if math.Abs(area) > 1 {
+		// Count any area above 0.99px is clickable
```

Comment:
```suggestion
		// Count any area above 0.99px as clickable
```

<hr>

I can infer what the comment says from the code, but not _why_ 0.99 is better than 1. Can you expand on that?

Also, are the units here really pixels?

Reply:
Yes, it's in pixels. The viewport is in pixels (what we compare to), and we're also returning a `Position` in pixels.

Do we want to click on something less than a pixel? :)

Reply:
Better?

```suggestion
		// We don't want to click on something less than a pixel.
```

Reply:
I see. I suppose if it's an area the units would be `px²` :smile:, but sure, that's fine.

Reply:
The "pixel" word in "We don't want to click on something less than a pixel." is `px².` It's clear from the context, isn't it? Otherwise, we'd need to have a 1d-mouse 😄